homeassistant/components/plex/media_player.py

```diff
@@ -553,27 +553,43 @@ def media_previous_track(self):
     def play_media(self, media_type, media_id, **kwargs):
         """Play a piece of media."""
         if not (self.device and "playback" in self._device_protocol_capabilities):
+            _LOGGER.debug(
+                "Client is not currently accepting playback controls: %s", self.name
+            )
             return
 
         src = json.loads(media_id)
-        library = src.get("library_name")
-        shuffle = src.get("shuffle", 0)
-
-        media = None
+        if media_type == "KEY" and isinstance(src, int):
```

Comment:
Doesn't media type have to be one of the existing media types in the base media_player integration?

https://developers.home-assistant.io/docs/entity_media_player#mediatype

Reply:
Apparently not as there doesn't seem to be any validation outside of the integration.

I'm intentionally adding a nonstandard/undocumented key here for troubleshooting purposes. The `play_media` service right now is two parts: 1) a Plex-specific search function to find the JSON-encoded media request and 2) the actual call to the Plex client to play that media. This PR intends to bypass step 1 as a failure point when testing.

Reply:
We don't always validate our architecture unfortunately. I think media type is only allowed to be one of those documented types. If we want something else it needs to be accepted in an architecture issue.

Reply:
As suggested in [this comment](https://github.com/home-assistant/core/pull/34550#issuecomment-618040897), I've changed the custom `media_content_type` value to "plex".